OSOE-92: ⭐ Publish Lombiq.VueJs on NuGet

sarahelsaig opened the pull request:
OSOE-92

sarahelsaig committed:
stylelint msbuild

sarahelsaig committed:
Fix markdown.

sarahelsaig committed:
submodules

sarahelsaig committed:
Remove ES6 Promise since nobody has to support IE any more.

sarahelsaig committed:
Remove gulp stuff from vue-app-css-pipeline.js.

sarahelsaig committed:
Merge remote-tracking branch 'origin/task/node-extensions' into issue/OSOE-92

# Conflicts:
#	Lombiq.OSOCE.sln
#	src/Modules/Lombiq.VueJs

sarahelsaig committed:
Merge remote-tracking branch 'origin/task/node-extensions' into issue/OSOE-92

# Conflicts:
#	Lombiq.OSOCE.sln

sarahelsaig committed:
submodules

sarahelsaig committed:
submodule

sarahelsaig committed:
submodules

sarahelsaig committed:
submodule

sarahelsaig committed:
Merge remote-tracking branch 'origin/task/node-extensions' into issue/OSOE-92

sarahelsaig committed:
submodules

sarahelsaig committed:
Fix root path for explore usage.

sarahelsaig committed:
code cleanup

sarahelsaig committed:
Handle other cases in run-pnpm-script.

sarahelsaig committed:
update submodule to get rid of MD violations.

sarahelsaig committed:
Merge remote-tracking branch 'origin/task/node-extensions' into issue/OSOE-92

# Conflicts:
#	test/Lombiq.UITestingToolbox

sarahelsaig committed:
Improve error handling.

sarahelsaig committed:
Need to check both because Windows and Linux resolves the directory symlink between the two differently.

sarahelsaig committed:
Update documentation.

sarahelsaig committed:
Code styling.

sarahelsaig committed:
Remove package-lock.json since we only use PNPM.

0liver commented on Lombiq.OSOCE.sln:
```diff
@@ -146,6 +146,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Lombiq.Hosting.Tenants.Feat
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Lombiq.Hosting.Tenants.FeaturesGuard.Tests.UI", "src\Modules\Lombiq.Hosting.Tenants\Lombiq.Hosting.Tenants.FeaturesGuard.Tests.UI\Lombiq.Hosting.Tenants.FeaturesGuard.Tests.UI.csproj", "{6EFBB112-DA07-4FAB-8D01-6300D7A06CA4}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Lombiq.VueJs", "src\Modules\Lombiq.VueJs\Lombiq.VueJs\Lombiq.VueJs.csproj", "{F9B7F1AD-1A35-48E2-93DE-5A9E17416F6D}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Lombiq.VueJs.Samples", "src\Modules\Lombiq.VueJs\Lombiq.VueJs.Samples\Lombiq.VueJs.Samples.csproj", "{9939AF29-9394-4E3A-8A94-F7995B40A5DA}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Lombiq.VueJs.Tests.UI", "src\Modules\Lombiq.VueJs\Lombiq.VueJs.Tests.UI\Lombiq.VueJs.Tests.UI.csproj", "{93B863A1-01EA-457E-B17D-54F3DFC5BF5B}"
+EndProject
```
While I would love our solution file to be ordered (in whatever way), it currently is not, and merging this with OSOE-84 will be a right-out pain.

Please use the same locations for the project and build configuration blocks as does the `dev` solution. I've already done that in [this commit](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/225/commits/de4512a655c5c64b2c7f028daa85160d4dcd2dec) - so you might just re-use that.

sarahelsaig replied:
I've reset this file to its state in `issue/OSOE-84` (excluding the changes in the solution items).

0liver replied:
I'm not sure your build will pass with that, since the other projects that aren't part of the solution in `task/node-extensions` still reference Gulp Extensions. Which is **not** part of the repo anymore.

sarahelsaig replied:
In this case, can we just stick what I had here for this PR?
During merge you can simply choose yours, because it's a superset of mine. No merge needed.

0liver replied:
If we merge it into OSOE-84, you're correct. OK then.

Commit:
Merge remote-tracking branch 'origin/task/node-extensions' into issue/OSOE-92

sarahelsaig committed:
Merge branch 'task/node-extensions' into issue/OSOE-92

sarahelsaig committed:
Reset Lombiq.OSOCE.sln to its state in OSOE-84.

sarahelsaig committed:
Merge branch 'issue/OSOE-92' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-92

sarahelsaig committed:
Revert "Reset Lombiq.OSOCE.sln to its state in OSOE-84."

This reverts commit 5fdc1a18c18f5097efab803c74124038860786b8.

Commit:
Merge branch 'task/node-extensions' into issue/OSOE-92

sarahelsaig committed:
update submodule

sarahelsaig committed:
using named exports

sarahelsaig committed:
Merge remote-tracking branch 'origin/task/node-extensions' into issue/OSOE-92

# Conflicts:
#	Lombiq.OSOCE.sln
#	src/Lombiq.OSOCE.Web/Lombiq.OSOCE.Web.csproj
#	src/Modules/Lombiq.VueJs
#	src/Utilities/Lombiq.NodeJs.Extensions

sarahelsaig committed:
Merge remote-tracking branch 'origin/task/node-extensions' into issue/OSOE-92

sarahelsaig committed:
Consolidate .Net packages.

sarahelsaig committed:
Reset sln file.

As discussed here: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/203#discussion_r1000589877

sarahelsaig committed:
Remove unnecessary file copy.

sarahelsaig committed:
Add software-terms and node dictionaries.

sarahelsaig committed:
branch selector

sarahelsaig committed:
Update assets and somewhat parallelize execution.

sarahelsaig committed:
Add missing nullish-coalescing operator.

sarahelsaig committed:
submodules

sarahelsaig committed:
submodules

sarahelsaig committed:
ESLint code cleanup.

sarahelsaig committed:
fast forward submodules

sarahelsaig committed:
merge with task/node-extensions

sarahelsaig committed:
Merge branch 'task/node-extensions' into issue/OSOE-92

0liver commented on .eslintrc.json:
```diff
@@ -0,0 +1,13 @@
+{
```
Delete this file, please.

sarahelsaig replied:
yes
![image](https://user-images.githubusercontent.com/4007293/198853592-297ebfa2-307e-4436-99fc-c9dbaf5e2337.png)

0liver replied:
No:

![image](https://user-images.githubusercontent.com/164360/199001126-ed84969b-e836-4902-9c3b-85d633bd4706.png)

sarahelsaig committed:
submodules

sarahelsaig committed:
delet this

sarahelsaig committed:
DRY rollup pipeline error handling.

sarahelsaig committed:
Use ^ in package.json

sarahelsaig committed:
Update lock file

sarahelsaig committed:
Hopefully fix fs.promises.mkdir().

Commit:
Replace old .eslintrc.json with new .eslintrc.js

[skip ci]

sarahelsaig committed:
Unhyphenate

sarahelsaig committed:
submodules

sarahelsaig committed:
Update scripts since "copy:assets" have been removed.

sarahelsaig committed:
Refactor executeFunctionByCommandLineArgument.

sarahelsaig committed:
Add "thenable".

see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenables

sarahelsaig committed:
Update branch selectors.

Commit:
Fix punctuation in submodule

[skip ci]

Commit:
Update submodules to merge commits
